python/cudaq/kernel/ast_bridge.py

```diff
@@ -2173,6 +2174,12 @@ def bodyBuilder(iterVal):
                 # Handle User-Custom Struct Constructor
                 cls, annotations = globalRegisteredTypes.getClassAttributes(
                     node.func.id)
+
+                if '__slots__' not in cls.__dict__:
+                    self.emitFatalError(
+                        f"Assigning to fields in data classes is not yet supported. The dataclass must be declared with @dataclass(slots=True) or @dataclasses.dataclass(slots=True).",
```

Comment:
Is this about assigning fields or adding fields ?

For example. If you have:
```python
    @dataclass(slots=True)
     class MyClass:
         x: int
         y: bool
```

Assigning values for `x` and `y` should be possible. Adding a new field, say `z`, shouldn't.